File: engine/engine-model/src/main/scala/tools/packet/guild/GuildNotice.scala
package tools.packet.guild

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GuildNotice(private var _guildId: Int, private var _notice: String) extends PacketInput {
  def guildId: Int = _guildId

  def notice: String = _notice

  override def opcode(): SendOpcode = SendOpcode.GUILD_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/guild/GuildNotice.java
package tools.packet.guild;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GuildNotice(Integer guildId, String notice) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GUILD_OPERATION;
   }
}